pkg/apis/batch/v1/defaults.go

```diff
@@ -26,6 +26,10 @@ func addDefaultingFuncs(scheme *runtime.Scheme) error {
 }
 
 func SetDefaults_Job(obj *batchv1.Job) {
+	if obj.Spec.Suspend == nil {
```

Comment:
yes, but since the field is mutable, this case might be different depending on what we decide here: https://github.com/kubernetes/kubernetes/pull/98727#discussion_r570467511

Reply:
I think the feature flag should prevent new uses of the feature, but not clobber existing uses of the feature. I would not expect it to be checked in the default setter, since the default should be the same either way.

Feature flags should add safety, not put the system in a less tested 3rd state when turned off.

Reply:
ok, then I guess we need to do something different then in the update strategy. Currently the PR sets the field back to nil on the following condition ``` !utilfeature.DefaultFeatureGate.Enabled(features.SuspendJob) && oldJob.Spec.Suspend == nil```, otherwise it allows the update. Since the field is defaulted here, all updates will be allowed because `oldJob.Spec.Suspend` will never be nil.

The other option is to not use a reference in the external type like `paused` in deployment and discussed in https://github.com/kubernetes/kubernetes/pull/98727/files#r571054554

Reply:
When the feature gate is on, we allow everything. When the feature gate is off, we have only a few combinations:

|old|new|what to do|
|-|-|-|
|nil|nil|we don't care, allow|
|nil|true/false|disallow ("prevent new uses of the feature")|
|true/false|nil|disallow ("not clobber existing uses of the feature" and "not put the system in a less tested 3rd state when turned off")|
|true/false|opposite|disallow|

So if the oldJob is nil, force newJob to stay nil. If the oldJob is non-nil, force newJob to be equal to oldJob. SG?

Updated strategy.go

Reply:
Note that defaulting also happens when apiserver loads an object from storage. That means that old.Spec.Suspended is never nil (only false)

For that reason, I still suggest that we just get rid of the pointer and use a plain boolean.

When the feature gate is off, would the controller do anything different? Should we just control in the apiserver by disallowing transitions?

Reply:
Again, the point of the feature gate is to restore the state of the cluster to a known working state, where .spec.suspended is always false. In that sense, I don't see a reason why we should prevent the true->false transition. We just need to prevent false->true.

Reply:
my only final issue with a non-pointer type is precedence. CronJob's suspend field is a boolptr too and has the same semantics from a API standpoint. it too is defaulted to false when loading from storage and the controller never sees a nil value.

@lavalamp for the final say? should booleans that default to false be bool pointers or plain bools? the controller using the field will never see a nil value thanks to defaulting.